src/runtime/propensity-server.ts

```diff
@@ -18,74 +18,78 @@ import {
   EventOriginator,
   EventParams,
 } from '../proto/api_messages';
+import {ClientEvent} from '../api/client-event-manager-api';
+import {Deps} from './deps';
+import {Fetcher} from './fetcher';
+import {PropensityScore} from '../api/propensity-api';
+import {Score} from '../api/propensity-api';
+import {ScoreDetail} from '../api/propensity-api';
 import {addQueryParam} from '../utils/url';
 import {adsUrl} from './services';
 import {analyticsEventToPublisherEvent} from './event-type-mapping';
 import {isBoolean, isObject} from '../utils/types';
 
+interface PropensityResponse {
+  header: {
+    ok: boolean;
+  };
+  scores: {
+    /* eslint-disable-next-line google-camelcase/google-camelcase */
+    error_message?: string;
+    product: string;
+    /* eslint-disable-next-line google-camelcase/google-camelcase */
+    score_type: number;
+    score: number;
+  }[];
+  error?: string;
+}
+
 /**
  * Implements interface to Propensity server
  */
 export class PropensityServer {
+  private readonly publicationId_: string;
+  clientId_: string | null = null;
+  version_ = 1;
+
   /**
    * Page configuration is known when Propensity API
    * is available, publication ID is therefore used
    * in constructor for the server interface.
-   * @param {!Window} win
-   * @param {!./deps.Deps} deps
-   * @param {!./fetcher.Fetcher} fetcher
    */
-  constructor(win, deps, fetcher) {
-    /** @private @const {!Window} */
-    this.win_ = win;
-    /** @private @const {!./deps.Deps} */
-    this.deps_ = deps;
-    /** @private @const {string} */
+  constructor(
+    private readonly win_: Window,
+    private readonly deps_: Deps,
+    private readonly fetcher_: Fetcher
+  ) {
     this.publicationId_ = this.deps_.pageConfig().getPublicationId();
-    /** @private {?string} */
-    this.clientId_ = null;
-    /** @private @const {!./fetcher.Fetcher} */
-    this.fetcher_ = fetcher;
```

Comment:
Does `this.fetcher_` need to be set here?

Reply:
Oh, the new code uses (`parameter properties`)[https://google.github.io/styleguide/tsguide.html#parameter-property-comments] to simplify the code. So `this.fetcher_` basically gets set on line 63 now